lib/context.js

```diff
@@ -9,6 +11,46 @@ class Context {
     Object.assign(this, request.query);
     this.log = request.log;
   }
+
+  cloudEventResponse(response) {
+    return new CloudEventResponse(response);
+  }
+
+}
+
+class CloudEventResponse {
+  #response;  // eslint-disable-line
+
+  constructor(response) {
+    this.#response = response;
+    if (!this.#response.headers) {
+      this.#response.headers = [];
+    }
+  }
+
+  version(version) {
+    this.#response.headers[Spec.version] = version;
+    return this;
+  }
+
+  id(id) {
+    this.#response.headers[Spec.id] = id;
+    return this;
+  }
+
+  type(type) {
+    this.#response.headers[Spec.type] = type;
+    return this;
+  }
+
+  source(source) {
+    this.#response.headers[Spec.source] = source;
+    return this;
+  }
+
+  response() {
```

Comment:
I wonder if this shouldn't be `data()` instead, since in the context of a cloud event, that's what this is.

Reply:
Ah right, that sounds good to me. I'll update shortly, thanks!

Reply:
Just looking at this a little closer, this function returns the response that is being decorated and is really just made available so that we can have everything on a single line, for example:
```js
const response = {                                                          
      message: context.cloudevent.data.message                                  
};                                                                           
return context.cloudEventResponse(response).version('0.3')                  
                                           .id('dummyid')                   
                                           .type('dev.ocf.js.type')         
                                           .source('dev/ocf/js/service')    
                                           .response();       
```
This could just as well be written as:
```js
const response = {                                                          
      message: context.cloudevent.data.message                                  
};                                                                           
context.cloudEventResponse(response).version('0.3')                  
                                           .id('dummyid')                   
                                           .type('dev.ocf.js.type')         
                                           .source('dev/ocf/js/service');    
return response;                                                               
````
The motivation for just decorating the response and not completely taking over the creation of the response is to enable users to add additional headers that they might need or do other things with the response that I've overlooked.